flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -67,6 +67,16 @@ public RexNodeConverter(RelBuilder relBuilder) {
 
 	@Override
 	public RexNode visitCall(CallExpression call) {
+		if (call.getFunctionDefinition().equals(BuiltInFunctionDefinitions.CAST)) {
```

Comment:
Why not dealing with this inside `visitScalarFunc`?

Reply:
OK, I will take code into `visitScalarFunc`.
The reason I haven't done this before is that `TypeLiteralExpression` can't be converted. I will do a little refactoring and move the code to `visitScalarFunc`.